liferay-frontend/liferay-portal
# LPS-117488 Consolidate Modal Selection UX

Okay... so this is maybe an over-reach to fix [Flickering styles in modals](https://issues.liferay.com/browse/LPS-117488). I had this cooking for a while on the side... I'm just sad @markocikos isn't around to greenlight this 😢 

**This PR:**
- Creates a new `openSelectionModal` utility on top of our current `openModal` to simplify and consolidate item selection
    - As a first step, this simplifies `openModal` and decouples it from the `selection` logic
- Replaces existing usages of `openModal` + `selectEventName` (since the selection logic lives now up in the chain)
- Replaces usages of `ItemSelectorDialog` with `openSelectionModal` (both single and multiple selection)
    - As a side-effect, we remove completely the `ItemSelectorDialog` component and the weird 3-step invocation pattern
- Replaces usages of `Liferay.Util.selectEntity` with `openSelectionModal`
    - As a side effect, this should allow us to also deprecate this API

**Benefits:**
- Consistent item selection throughout DXP. This was a key item in `7.2` that we're in risk of regressing in `7.3`, which is what really pushed me to try to get it in.
- Reduced number of patterns to do the same thing (easier to document and enforce in the future)

**How to test it:**
After this change, this can be tested almost everywhere we "select" things, though the ones in the issue report should be enough to see the changes:

- Go to Global Menu > Content Dashboard
- Filter by Author or Subtype

- Go to Web Content
- Create a Web Content
- Select a display page template

After the change, the selection would be achieved through a Clay Modal (animation and everything) and styles won't flicker since all the css classes are in place and content is not shown until the iframe is done loading.

Going to mention some teams here today and personally tomorrow so we can validate this, but ideally we should push this asap to have enough time to stabilize.
- Hey @liferay-lima, hope you're okay with us removing the `ItemSelectorDialog` component. I know we discussed in the past where this should live, so maybe this makes sense to you? We're still using all the `ItemSelector` infra, just opening them through other means.
- Hey @bryceosterhaus, it was here where I "noticed" the iframe performance regression I mentioned at [Improve Iframe loading perceived performance](https://issues.liferay.com/browse/LPS-118805). This could be just subjective, but while I was changing modals and checking they worked, they felt slower to me. Could you maybe identify some modals here and try before and after to get a feeling for yourself?
- Hey @john-co, @PablitoBonito, I ran a couple of test suites at https://github.com/jbalsas/liferay-portal/pull/2226#issuecomment-674895777. Most of the failures are apparently just iframe locators. Could we prioritize this in a way that we can get this **merged asap**?

Thanks everyone! 🤞 

/cc @matuzalemsteles, @diegonvs, @brunobasto, could you give this a try, check test failures and see how safe this feels? :)
